src/main/java/org/jabref/logic/cleanup/URLCleanup.java

```diff
@@ -0,0 +1,61 @@
+package org.jabref.logic.cleanup;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jabref.model.FieldChange;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.StandardField;
+
+/**
+ * Checks whether URL exists in note field, and stores it under url field .
+ */
+public class URLCleanup implements CleanupJob {
+
+    private static final Field NOTE_FIELD = StandardField.NOTE;
+    private static final Field URL_FIELD = StandardField.URL;
+
+    @Override
+    public List<FieldChange> cleanup(BibEntry entry) {
+        List<FieldChange> changes = new ArrayList<>();
+
+        String NoteFieldValue = entry.getField(NOTE_FIELD).orElse(null);
+        String urlRegex = (
+                "(?i)\\b((?:https?://|www\\d{0,3}[.]|[a-z0-9.\\-]+[.]"
```

Comment:
Can you please add a comment/link on the regex, e.g where you got it from

Reply:
This was originally fetched from [stackoverflow](https://stackoverflow.com/questions/28185064/python-infinite-loop-in-regex-to-match-url?fbclid=IwAR2aH5PWTwfzilmnfWYDIHmr4IsLRdWkIzNy5dQVggINVZ3n11cEtWmBsM4), but there were some modifications in order for it to be functional. Shall I comment that link in my code?

Reply:
Yes, and the modifications you did